les/utils/expiredvalue.go

```diff
@@ -124,6 +124,47 @@ func (e *ExpiredValue) SubExp(a ExpiredValue) {
 	}
 }
 
+// LinearExpiredValue is very similiar with the expiredValue which the value
+// will continuously expired. But the different part is it's expired linearly.
+type LinearExpiredValue struct {
```

Comment:
Maybe it would be nicer to integrate the expiration rate into this struct because it doesn't look so good to always do the division externally.

Reply:
done